test/util/TruthTable.ts

```diff
@@ -1,159 +1,188 @@
-import type * as RDF from 'rdf-js';
-
-import { termToString } from 'rdf-string';
-import type { GeneralEvaluationConfig } from './generalEvaluation';
+import { stringToTermPrefix } from './Aliases';
 import { generalEvaluate } from './generalEvaluation';
+import type { TestArgument } from './utils';
 import { template } from './utils';
 
-// Maps short strings to longer RDF term-literals for easy use in making
-// evaluation tables.
-// Ex: { 'true': '"true"^^xsd:boolean' }
-export type AliasMap = Record<string, string>;
-
-// Maps short strings to longer RDF term-objects for easy use in making
-// evaluation tables.
-// Ex: { 'true': RDFDM.literal("true", RDF.namedNode(DT.XSD_BOOLEAN))}
-export type ResultMap = Record<string, RDF.Term>;
-
-// A series of tests in string format.
-// A difference is made between erroring and non-erroring tests-cases because
-// the testing framework requires it, and it allows for some reduction in
-// boilerplate when errorhandling is similar between operators.
-// Some mappings are passed to translate between shorthand aliases and their
-// full SPARQL representations.
-//
-// It's format is
-//  - line separation between rows (= test cases)
-//  - space separation between args
-//  - '=' separation between the args and the result
-//
-// Ex: `
-// true true = false
-// false false = false
-// `
-export interface IEvaluationConfig {
-  op: string;
-  arity: number;
-  aliasMap: AliasMap;
-  resultMap: ResultMap;
-  notation: Notation;
-  generalEvaluationConfig?: GeneralEvaluationConfig;
-}
-export type EvaluationTable = IEvaluationConfig & {
-  table?: string;
-  errorTable?: string;
-};
-
 export enum Notation {
   Infix,
   Prefix,
+  Suffix,
   Function,
 }
 
-// Given the definition for an evaluation table, test all it's test cases.
-export function testTable(definition: EvaluationTable): void {
-  const tTable = (definition.arity === 2) ?
-    new BinaryTable(definition, BinaryTableParser) :
-    new UnaryTable(definition, UnaryTableParser);
+type Row = [string, string, string] | [string, string] | string[];
```

Comment:
Any difference with this?
```suggestion
type Row = string[];
```

Reply:
Not really. I kept it because I thought it would show the difference between the 3 more but it doesn't really matter.

Reply:
I do prefer it like this as well, highlights the three cases.

Reply:
AFAIK it won't make any difference for autocompletion within IDEs, but no strong opinion eitherway.

Reply:
It doesn't indeed. It's committed like this now, let's keep it as such.